java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/prometheus/PrometheusSinkFunction.java

```diff
@@ -149,7 +159,15 @@ public void invoke(RowData row, Context context) throws Exception {
         for (int i = 0; i < columnInfos.size(); i++) {
             updateMetric(row, i);
         }
-        pushGateway.pushAdd(registry, jobName, extraLabels);
+        try {
+            retryUntilTimeout(() -> pushGateway.pushAdd(registry, jobName, extraLabels));
+        } catch (Exception e) {
+            LOG.warn(
```

Comment:
It seems simpler to just log the message inside `retryUntilTimeout`.

Reply:
The parameter of `log.warn` must start with a `String`, like `void warn(String var1, Throwable var2)`. Thus an additional warning message is needed.